Sources/MapboxMaps/Location/Puck/PuckManager.swift

```diff
@@ -5,6 +5,7 @@ import UIKit
 internal protocol PuckManagerProtocol: AnyObject {
     var puckType: PuckType? { get set }
     var puckBearingSource: PuckBearingSource { get set }
+    var puckBearingEnabled: Bool { get set }
 }
 
 internal final class PuckManager: PuckManagerProtocol {
```

Comment:
Needs tests

Reply:
Update these:
* `PuckManagerTests.testInitialPropertyValues`
* `PuckManagerTests.testSetPuckTypeToPuck2D`
* `PuckManagerTests.testSetPuckTypeToPuck3D`

Add `PuckManagerTests.testSettingPuckBearingEnabledWhenPuckTypeIsNonNil` which should be similar to `.testSettingPuckBearingSourceWhenPuckTypeIsNonNil`

Reply:
Still some changes I'd like to see here —

`PuckManagerTests.testSetPuckTypeToPuck2D`: at the beginning of the test, initialize `puckManager.puckBearingEnabled` to a random value (you can write `= .random()` since it's just a `Bool`. Then, in the first group of assertions, assert that the value was propagated to `puck` as expected. Then in the last group of assertions, reset `$bearingSource.setStub` and verify that it doesn't get invoked again when the same puck is set again.

`PuckManagerTests.testSetPuckTypeToPuck3D` should work the same way.